src/main/java/de/tum/in/www1/artemis/repository/BuildPlanRepository.java

```diff
@@ -22,13 +23,25 @@
             """)
     Optional<BuildPlan> findByProgrammingExercises_IdWithProgrammingExercises(@Param("exerciseId") long exerciseId);
 
+    default BuildPlan findByProgrammingExercises_IdWithProgrammingExercisesElseThrow(long exerciseId) {
```

Comment:
Please stay consistent with your final declarations. For the other method you added them, so please do so here, too.